x-pack/plugins/transform/public/app/sections/create_transform/components/group_by_list/popover_form.tsx

```diff
@@ -263,6 +280,31 @@ export const PopoverForm: React.FC<Props> = ({ defaultData, otherAggNames, onCha
           </>
         </EuiFormRow>
       )}
+      {isGroupByDateHistogram(defaultData) && timeZone ? (
+        <EuiFormRow
+          error={
+            !timeZoneValid && [
+              i18n.translate('xpack.transform.groupBy.popoverForm.timeZoneError', {
+                defaultMessage: 'Invalid time zone.',
+              }),
+            ]
+          }
```

Comment:
`isInvalid` prop takes care of showing/hiding errors
```suggestion
          error={i18n.translate('xpack.transform.groupBy.popoverForm.timeZoneError', {
                defaultMessage: 'Invalid time zone.',
              })
```